builder/amazon/ebsvolume/step_tag_ebs_volumes.go

```diff
@@ -36,34 +38,85 @@ func (s *stepTagEBSVolumes) Run(ctx context.Context, state multistep.StateBag) m
 	state.Put("ebsvolumes", volumes)
 
 	if len(s.VolumeMapping) > 0 {
-		ui.Say("Tagging EBS volumes...")
+		// If run_volume_tags were set in the template any attached EBS
+		// volume will have had these tags applied when the instance was
+		// created. We now need to remove these tags to ensure only the EBS
+		// volume tags are applied (if any)
+		toDeTag := map[string][]*ec2.Tag{}
+		if config.VolumeRunTags.IsSet() {
+			ui.Say("Removing any tags applied to EBS volumes when the source instance was created...")
 
+			ui.Message("Compiling list of existing tags to remove...")
+			existingTags, err := config.VolumeRunTags.EC2Tags(s.Ctx, *ec2conn.Config.Region, state)
+			if err != nil {
+				err := fmt.Errorf("Error generating list of tags to remove: %s", err)
+				state.Put("error", err)
+				ui.Error(err.Error())
+				return multistep.ActionHalt
+			}
+			existingTags.Report(ui)
+
+			// Generate the map of volumes and associated tags to delete.
+			// Looping over the instance block device mappings allows us to
+			// obtain the volumeId
+			for _, mapping := range s.VolumeMapping {
+				for _, v := range instance.BlockDeviceMappings {
+					if *v.DeviceName == mapping.DeviceName {
+						toDeTag[*v.Ebs.VolumeId] = existingTags
+					}
+				}
+			}
+
+			// Delete the tags
+			for volumeId, existingTags := range toDeTag {
+				ui.Message(fmt.Sprintf("Deleting existing 'run_volume_tags' on EBS Volume: %s", volumeId))
+				_, err = ec2conn.DeleteTags(&ec2.DeleteTagsInput{
+					Resources: aws.StringSlice([]string{volumeId}),
```

Comment:
here, if we pass the volume ids as a list, will it allow not doing these delete requests in a for loop ?
Eg;
```golang
&ec2.DeleteTagsInput{
	Resources: volumeIds,
```

Reply:
Ah! Good catch! All the tags will be the same here so we can remove them with one API call.

Reply:
nice 🙂